### ISSUE TITLE
Latest build of libunistring.dll for clang64 doesn't provide "locale_charset" function.

### ISSUE THREAD

**Issue by elohim-meth** [msys2/MINGW-packages#17450]
### Description / Steps to reproduce the issue

Trying to link a program that uses "locale_charset" function gives an error  "ld.lld: error: undefined symbol: locale_charset".
DLL Export Viewer on /msys/clang64/bin/libunistring-5.dll shows no such function, but it exists in /msys/mingw64/bin/libunistring-5.dll 

### Expected behavior

No error when linking

### Actual behavior

Error when linking

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [X] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by Biswa96** [comment]
> Trying to link a program that uses "locale_charset" function gives an error 

Could you provide steps to reproduce the issue? locale_charset is also exported from libiconv and libcharset. And libiconv is a dependency of libunisrting.

**Comment by elohim-meth** [comment]
> locale_charset is also exported from libiconv and libcharset. And libiconv is a dependency of libunisrting.
Yes it is exported from libiconv, but it's a different function and libunistring has its own one. 
Yes libiconv is a dependency of libunistring, but linking to libunistring doesn't autolink to libiconv.

**Comment by elohim-meth** [comment]
main.cpp:
```
#include <iostream>
#include <uniconv.h>

using namespace std;
int main()
{
    auto *ustr = locale_charset();
    cout << "Charset name: " << ustr << endl;
}
```
CMakeLists.txt
```
cmake_minimum_required(VERSION 3.16)

project(test LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

add_executable(${PROJECT_NAME} main.cpp)

find_library(LIBUNISTRING NAMES unistring)
if (LIBUNISTRING)
    message(STATUS "Found libunistring: ${LIBUNISTRING}")
    target_link_libraries(${PROJECT_NAME} PRIVATE ${LIBUNISTRING})
else()
    message(FATAL_ERROR "Required libunistring not found")
endif()
```
